wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -94,6 +95,9 @@ class WireJsonAdapterFactory private constructor(
     if (rawType == AnyMessage::class.java) {
       return AnyMessageJsonAdapter(moshi, typeUrlToAdapter)
     }
+    if (rawType == Duration::class.java) {
```

Comment:
This will require users be using AGP 4.0 and have `coreLibraryDesugaring true`. You might want to wrap this in try/catch for NoClassDefFoundError and ignore so it works on AGP 3.x and/or AGP 4 without core library desugaring.